sdk/provisioning/Azure.Provisioning/tests/SampleTests.cs

```diff
@@ -0,0 +1,327 @@
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.OperationalInsights;
+using Azure.Provisioning.Primitives;
+using Azure.Provisioning.Resources;
+using Azure.Provisioning.Roles;
+using Azure.Provisioning.Storage;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using NUnit.Framework;
+
+namespace Azure.Provisioning.Tests;
+
+internal class SampleTests(bool async)
+    : ProvisioningTestBase(async /*, skipTools: true, skipLiveCalls: true /**/)
+{
+    [RecordedTest]
+    public async Task SimpleDeploy()
+    {
+        BlobService? blobs = null;
+        BicepOutput? endpoint = null;
+
+        await using Trycep test = CreateBicepTest();
+        await test.Define(
+            ctx =>
+            {
+                BicepParameter location = BicepParameter.Create<string>(nameof(location), BicepFunction.GetResourceGroup().Location);
+
+                // Create a storage account and blob resources
+                StorageAccount storage = StorageResources.CreateAccount(nameof(storage));
+                storage.Location = location;
+                blobs = new(nameof(blobs)) { Parent = storage };
+
+                // Grab the endpoint
+                endpoint = BicepOutput.Create<string>("blobs_endpoint", storage.PrimaryEndpoints.Value!.BlobUri);
+            })
+        .Compare(
+            """
+            param location string = resourceGroup().location
+
+            resource storage 'Microsoft.Storage/storageAccounts@2023-01-01' = {
+                name: take('storage${uniqueString(resourceGroup().id)}', 24)
+                kind: 'StorageV2'
+                location: location
+                sku: {
+                    name: 'Standard_LRS'
+                }
+                properties: {
+                    allowBlobPublicAccess: false
+                    isHnsEnabled: true
+                }
+            }
+
+            resource blobs 'Microsoft.Storage/storageAccounts/blobServices@2023-01-01' = {
+                name: 'default'
+                parent: storage
+            }
+
+            output blobs_endpoint string = storage.properties.primaryEndpoints.blob
+            """)
+        .Lint()
+        .ValidateAndDeployAsync(
+            async deployment =>
+            {
+                // Create a client
+                BlobServiceClient storage = InstrumentClient(
+                    deployment.CreateClient(
+                        blobs!,
+                        TestEnvironment.Credential,
+                        InstrumentClientOptions(new BlobClientOptions())));
+
+                // Make sure the output matched correctly
+                Assert.AreEqual(storage.Uri.AbsoluteUri, endpoint!.Value.Value);
+
+                // Make a service call and verify it doesn't throw
+                Response<BlobServiceProperties> result = await storage.GetPropertiesAsync();
+                Assert.AreEqual(200, result.GetRawResponse().Status);
+            });
```

Comment:
Or go end to end by creating the Storage account and constructing a `BlobServiceClient` directly from the deployed resources.